gadopt/utility.py

```diff
@@ -475,3 +477,56 @@ def node_coordinates(function):
     return [
         Function(func_space).interpolate(coords).dat.data for coords in mesh_coords
     ]
+
+
+def assign_1d_profile(function: Function, one_d_filename: str, cartesian: bool, r1d=None):
+    """
+    Assign a one-dimensional profile to a Function `q` from a file.
+
+    The function reads a one-dimensional radial viscosity profile together with the
+    radius input for the profile, from a file, broadcasts the read array to all
+    processes, and then interpolates this array onto the function space of `q`.
+
+    Args:
+        q (Function): The function onto which the 1D profile will be assigned.
+        one_d_filename (str): The path to the file containing the 1D radial viscosity profile.
+        cartesian (bool): Is the upward direction along z/y(cartesian=True) or radial (cartesian=False)
+
+    Returns:
+        None: It directly assigns the input function `function`.
+
+    Note:
+        - Note the cartesian flag
+        - This is designed to read a file with one process and distribute in parallel with MPI.
+        - The input file should contain an array of radius and an array of viscosity values, seperated by comma.
+        - Make sure radius in the file is consistent with the mesh.
+    """
+    # find the mesh
+    mesh = extract_unique_domain(function)
+
+    visc = None
+    rshl = None
+
+    # read the input file
+    if mesh.comm.rank == 0:
+        # The root process reads the file
+        rshl, visc = np.loadtxt(one_d_filename, unpack=True, delimiter=",")
+
+    # Broadcast the entire 'visc' array to all processes
+    visc = mesh.comm.bcast(visc, root=0)
+    # Similarly, broadcast 'rshl' if needed (assuming all processes need it)
+    rshl = mesh.comm.bcast(rshl, root=0)
+
+    # Coordinates
+    X = SpatialCoordinate(mesh)
+
+    if cartesian:
+        # if in a two-dimensional problem: y otherwise z is the direction
+        upward_coord = X[1] if len(X) == 2 else X[2]
+    else:
+        upward_coord = sqrt(X**2)
+
+    rad = Function(function.function_space()).interpolate(upward_coord)
+
+    averager = LayerAveraging(mesh, rshl if mesh.layers is None else None, cartesian=cartesian)
+    averager.extrapolate_layer_average(function, interp1d(rshl, visc, fill_value="extrapolate")(averager.get_layer_average(rad)))
```

Comment:
It would be more readable if the interpolator object was defined outside of the method call. Otherwise, you could add `.__call__`.

Reply:
[SciPy's doc](https://docs.scipy.org/doc/scipy/reference/generated/scipy.interpolate.interp1d.html#scipy.interpolate.interp1d) states 

> This class is considered legacy and will no longer receive updates. This could also mean it will be removed in future SciPy versions.

Suggestions for replacement can be found [here](https://docs.scipy.org/doc/scipy/tutorial/interpolate/1D.html#tutorial-interpolate-1dsection).

Reply:
Changing to `np.interp`. Thanks.